src/txn/txn.c

```diff
@@ -921,9 +923,20 @@ __txn_resolve_prepared_op(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit,
           true);
 
         if (ret == 0)
-            WT_ERR(__txn_append_hs_record(session, hs_cursor, &op->u.op_row.key,
-              ((WT_CURSOR_BTREE *)(*cursorp))->ref->page, upd, commit, &fix_upd, &upd_appended));
-        else
+            WT_ERR(__txn_append_hs_record(session, hs_cursor, &op->u.op_row.key, cbt->ref->page,
+              upd, commit, &fix_upd, &upd_appended));
+        else if (ret == WT_NOTFOUND && !commit) {
+            /*
+             * Allocate a tombstone so that when we reconcile the update chain we don't copy the
+             * prepared cell, which is now associated with a rolled back prepare, and instead write
+             * nothing.
+             */
+            WT_ERR(__wt_upd_alloc_tombstone(session, &tombstone, &size));
+            /* Apply the tombstone to the row. */
+            WT_WITH_BTREE(session, op->btree, ret = __wt_row_modify(cbt, &cbt->iface.key, NULL,
```

Comment:
Does this work?

Reply:
Yeah I'll push up a test that demonstrates it.

Reply:
But to be clear it works as `__wt_row_modify` checking for prepare conflicts.